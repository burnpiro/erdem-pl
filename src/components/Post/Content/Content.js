// @flow
import React from 'react';
import rehypeReact from 'rehype-react';
import styles from './Content.module.scss';
import Meta from '../Meta';
import XAIInfidelity from '../../Diagrams/XAIInfidelity';
import RNNProcess from '../../Diagrams/RNNProcess/RNNProcess';
import RNNWithAttention from '../../Diagrams/RNNWithAttention/RNNWithAttention';
import ImageWithAttention from '../../Diagrams/ImageWithAttention/ImageWithAttention';
import InputPositionEmbeddingSin from '../../Diagrams/InputPositionEmbeddingSin/InputPositionEmbeddingSin';
import AttentionLayer from '../../Diagrams/AttentionLayer/AttentionLayer';

type Props = {
  body: string,
  htmlAst: string,
  title: string,
  date: string,
  readTime?: {
    text: string,
    minutes: number,
  },
};

const renderAst = new rehypeReact({
  createElement: React.createElement,
  components: {
    'xai-infidelity': XAIInfidelity,
    'rnn-process': RNNProcess,
    'rnn-with-attention': RNNWithAttention,
    'image-with-attention': ImageWithAttention,
    'sin-position-embedding': InputPositionEmbeddingSin,
    'attention-layer': AttentionLayer,
  },
}).Compiler;

const Content = ({ body, htmlAst, title, date, readTime }: Props) => (
  <div className={styles['content']}>
    <h1 className={styles['content__title']}>{title}</h1>
    <Meta date={date} readTime={readTime} />
    <div
      className={styles['content__body']}
      // dangerouslySetInnerHTML={{ __html: body }}
    >
      {renderAst(htmlAst)}
    </div>
  </div>
);

export default Content;
