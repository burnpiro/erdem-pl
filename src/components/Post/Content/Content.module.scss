@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.content {
  max-width: $layout-post-single-width;
  padding: 0 15px;
  margin: 0 auto;

  &__title {
    font-size: $typographic-base-font-size * 2;
    max-width: $layout-post-width;
    margin-left: auto;
    margin-right: auto;
    font-weight: 600;
    text-align: center;
    @include line-height(1.65);
    @include margin-top(1);
    @include margin-bottom(0);
  }

  &__body {
    & blockquote {
      font-style: italic;
      text-align: center;
      @include margin-top(0);
      background: #ededed;
      color: #111111;
      position: relative;
      border-left: 8px solid $color-secondary;
      @include padding(1, 0.5, 1, 1);

      &:before {
        content: '\201C';
        color: $color-secondary;
        font-size: $typographic-base-font-size * 4;
        position: absolute;
        @include top(1);
        @include left(0);
      }

      &:after {
        content: '';
      }

      & p {
        max-width: $layout-post-width;
        @include margin-top(0);
        @include margin-bottom(0);
        @include line-height(1.5);
      }

      & ul {
        text-align: left;
      }

      & a {
        font-weight: bold;
      }

      //& span {
      //  display: block;
      //  color: #333333;
      //  font-style: normal;
      //  font-weight: bold;
      //  margin-top: 1em;
      //}
    }
    & figure {
      @include margin-bottom(1);
    }

    & a {
      text-decoration: underline;
    }

    & div {
      max-width: $layout-post-width;
      margin-left: auto;
      margin-right: auto;
    }

    & img {
      max-width: 100%;
      cursor: pointer;

      &[alt$='full'] {
        max-width: none;
        max-height: none;
        width: 100vw !important;
        margin-left: calc((-100vw + 100%) / 2) !important;
      }
    }
  }
}

@include breakpoint-md {
  .content {
    padding: 0;

    &__title {
      font-size: $typographic-base-font-size * 3;
      @include line-height(2.25);
      @include margin-top(2.25);
      @include margin-bottom(1.5);
    }

    &__body {
      font-size: $typographic-base-font-size * 1.125;
      @include line-height(1.125);
      @include margin-bottom(1.125);

      & p {
        font-size: $typographic-base-font-size * 1.125;
        @include line-height(1.125);
        @include margin-bottom(1.125);
      }
    }
  }
}
