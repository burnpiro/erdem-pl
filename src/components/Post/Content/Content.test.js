// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Content from './Content';

describe('Content', () => {
  it('renders correctly', () => {
    const props = {
      title: 'test',
      body: '<p>test</p>',
      htmlAst: { children: [], data: { quirksMode: false }, type: 'root' },
      date: '2019-07-10',
      readTime: {
        text: '5 minutes',
        minutes: 5,
      },
    };

    const tree = renderer.create(<Content {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
