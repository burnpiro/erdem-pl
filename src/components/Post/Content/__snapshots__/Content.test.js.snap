// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<div
  className="content"
>
  <h1
    className="content__title"
  >
    test
  </h1>
  <div
    className="meta"
  >
    <span
      className="meta__date"
    >
      Published 
      10 Jul 2019
    </span>
    <span
      className="meta__dot"
    >
       Â· 
    </span>
    <span
      className="readTime"
    >
      <img
        alt="5min coffee icon"
        className="readTime__icon"
        src="file"
      />
      <img
        alt="5min coffee icon"
        className="readTime__icon"
        src="file"
      />
      <span
        className="readTime__text"
      >
        5 minutes
      </span>
    </span>
  </div>
  <div
    className="content__body"
  >
    <div />
  </div>
  <div
    className="citation"
  >
    <h3>
      Citation
    </h3>
    <blockquote
      className="citation__text"
    >
      Test name, (Jul 2019). "test". 
      <a
        href="http://localhost/2020/02/understanding-region-of-interest-ro-i-pooling"
      >
        http://localhost/2020/02/understanding-region-of-interest-ro-i-pooling
      </a>
    </blockquote>
    <b>
      or
    </b>
    <pre
      className="citation__code"
      data-language="javascript"
    >
      <code>
        @article{erdem2019understandingRegionOfInterestRoIPooling,
    title   = "test",
    author  = "Test name",
    journal = "http://localhost",
    year    = "2019",
    month   = "Jul",
    url     = "http://localhost/2020/02/understanding-region-of-interest-ro-i-pooling"
}

      </code>
    </pre>
  </div>
</div>
`;
